[bug] can't compile tauri for development

### Describe the bug

cannot find type `Command` in module `crate::api::process`

failed to resolve: could not find `Encoding` in `process`

problem occurred on latest Tauri v1.7.0 while I'm creating a new tauri project on Win11 using command `pnpm create tauri-app`

same problem as #9728, and I've tried to update cargo dependencies, but nothing helpful

### Reproduction

_No response_

### Expected behavior

_No response_

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 126.0.2592.81
    ✔ MSVC: Visual Studio Build Tools 2022
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.15.0
    - pnpm: 9.5.0-beta.0
    - npm: 10.8.1

[-] Packages
    - tauri [RUST]: 1.7.0
    - tauri-build [RUST]: 1.5.3
    - wry [RUST]: 0.24.10
    - tao [RUST]: 0.16.9
    - @tauri-apps/api [NPM]: 1.6.0
    - @tauri-apps/cli [NPM]: 1.6.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```


### Stack trace

```text
Compiling tauri v1.7.0
error[E0412]: cannot find type `Command` in module `crate::api::process`
   --> C:\Users\*****\scoop\persist\rustup\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-1.7.0\src\endpoints\shell.rs:244:41
    |
244 | ) -> super::Result<crate::api::process::Command> {
    |                                         ^^^^^^^ not found in `crate::api::process`
    |
help: consider importing this struct
    |
7   + use std::process::Command;
    |
help: if you import `Command`, refer to it directly
    |
244 - ) -> super::Result<crate::api::process::Command> {
244 + ) -> super::Result<Command> {
    |

error[E0433]: failed to resolve: could not find `Encoding` in `process`
   --> C:\Users\*****\scoop\persist\rustup\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-1.7.0\src\endpoints\shell.rs:304:50
    |
304 |     if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
    |                                                  ^^^^^^^^ could not find `Encoding` in `process`
    |
help: consider importing this struct
    |
7   + use encoding_rs::Encoding;
    |
help: if you import `Encoding`, refer to it directly
    |
304 -     if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
304 +     if let Some(encoding) = Encoding::for_label(encoding.as_bytes()) {
    |

   Compiling vswhom v0.1.0
   Compiling embed-resource v2.4.2
   Compiling tauri-winres v0.1.1
   Compiling tauri-build v1.5.3
Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `tauri` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```


### Additional context

_No response_

--- a/.changes/shell-execute.md
+++ b/.changes/shell-execute.md
@@ -1,0 +1,5 @@
+---
+"tauri": "patch:bug"
+---
+
+Fix compilation error when `shell-execute` or `shell-sidecar` features are not active

--- a/core/tauri/src/endpoints.rs
+++ b/core/tauri/src/endpoints.rs
@@ -40,6 +40,7 @@ mod window;
 /// The context passed to the invoke handler.
 pub struct InvokeContext<R: Runtime> {
   pub window: Window<R>,
+  #[allow(dead_code)]
   pub config: Arc<Config>,
   pub package_info: PackageInfo,
 }

--- a/core/tauri/src/endpoints/notification.rs
+++ b/core/tauri/src/endpoints/notification.rs
@@ -18,6 +18,7 @@ const PERMISSION_GRANTED: &str = "granted";
 const PERMISSION_DENIED: &str = "denied";
 
 #[derive(Debug, Clone)]
+#[allow(dead_code)]
 pub enum SoundDto {
   Default,
   Custom(String),
@@ -49,6 +50,7 @@ impl<'de> Deserialize<'de> for SoundDto {
 
 /// The options for the notification API.
 #[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
 pub struct NotificationOptions {
   /// The notification title.
   pub title: String,

--- a/core/tauri/src/endpoints/shell.rs
+++ b/core/tauri/src/endpoints/shell.rs
@@ -110,7 +110,8 @@ impl Cmd {
     let encoding = options
       .encoding
       .as_ref()
-      .and_then(|encoding| crate::api::process::Encoding::for_label(encoding.as_bytes()));
+      .and_then(|encoding| encoding_rs::Encoding::for_label(encoding.as_bytes()));
+
     let command = prepare_cmd(&context, &program, args, options)?;
 
     let mut command: std::process::Command = command.into();
@@ -236,6 +237,7 @@ impl Cmd {
   }
 }
 
+#[cfg(any(shell_execute, shell_sidecar))]
 fn prepare_cmd<R: Runtime>(
   context: &InvokeContext<R>,
   program: &String,
@@ -301,7 +303,7 @@ fn prepare_cmd<R: Runtime>(
     command = command.env_clear();
   }
   if let Some(encoding) = &options.encoding {
-    if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
+    if let Some(encoding) = encoding_rs::Encoding::for_label(encoding.as_bytes()) {
       command = command.encoding(encoding);
     } else {
       return Err(anyhow::anyhow!(format!("unknown encoding {encoding}")));

--- a/tooling/bundler/src/bundle/macos/sign.rs
+++ b/tooling/bundler/src/bundle/macos/sign.rs
@@ -305,23 +305,21 @@ pub fn notarize(
       log::info!(action = "Notarizing"; "{}", log_message);
       staple_app(app_bundle_path)?;
       Ok(())
-    } else {
-      if let Ok(output) = Command::new("xcrun")
-        .args(["notarytool", "log"])
-        .arg(&submit_output.id)
-        .notarytool_args(&auth)
-        .output_ok()
-      {
-        Err(
-          anyhow::anyhow!(
-            "{log_message}\nLog:\n{}",
-            String::from_utf8_lossy(&output.stdout)
-          )
-          .into(),
+    } else if let Ok(output) = Command::new("xcrun")
+      .args(["notarytool", "log"])
+      .arg(&submit_output.id)
+      .notarytool_args(&auth)
+      .output_ok()
+    {
+      Err(
+        anyhow::anyhow!(
+          "{log_message}\nLog:\n{}",
+          String::from_utf8_lossy(&output.stdout)
         )
-      } else {
-        Err(anyhow::anyhow!("{log_message}").into())
-      }
+        .into(),
+      )
+    } else {
+      Err(anyhow::anyhow!("{log_message}").into())
     }
   } else {
     Err(anyhow::anyhow!("failed to parse notarytool output as JSON: `{output_str}`").into())

--- a/tooling/bundler/src/bundle/path_utils.rs
+++ b/tooling/bundler/src/bundle/path_utils.rs
@@ -18,6 +18,7 @@ pub struct DirOpts {
 pub struct FileOpts {
   pub overwrite: bool,
   pub skip: bool,
+  #[allow(dead_code)]
   pub buffer_size: usize,
 }
 

--- a/tooling/bundler/src/bundle/windows/sign.rs
+++ b/tooling/bundler/src/bundle/windows/sign.rs
@@ -42,6 +42,8 @@ impl Settings {
     }
   }
 }
+
+#[allow(dead_code)]
 pub struct SignParams {
   pub product_name: String,
   pub digest_algorithm: String,

